### PR TITLE
feat: add popup information

### PR DESCRIPTION
**Pull request by be-ndee**
- fix bug
- add documentation
- resolve merge conflicts with master
- improve code

### CODE REVIEW

File: lua/nvim-tree/actions/file-popup.lua [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  end
+end
+
+function M.show_file_info()
+  M.close_popup()
+
+  local node = require'nvim-tree.lib'.get_node_at_cursor()
+  if not node or not node.absolute_path then
+    return
+  end
```
Comment (kyazdani42) [inline lua/nvim-tree/actions/file-popup.lua:57]: you can remove this, the function can take the `node` as parameter

**Review by kyazdani42 — APPROVED** [review]

**Comment by kyazdani42** [comment]
thanks :)

File: lua/nvim-tree/actions/file-popup.lua [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+local function setup_window(lines)
+  local max_width = vim.fn.max(vim.tbl_map(function(n) return #n end, lines))
+  winnr = a.nvim_open_win(0, false, {
+    col = 1,
+    row = 1,
+    relative = "cursor",
+    width = max_width + 1,
+    height = #lines,
+    border = 'shadow',
```
Comment (gegoune) [inline lua/nvim-tree/actions/file-popup.lua:33]: This (border) feels bit too opinionated, could it be made configurable maybe?

Comment (kyazdani42) [reply lua/nvim-tree/actions/file-popup.lua:33]: this is just a first draft, this feature can be extended as needed. Adding a config for `actions.file_popup = { ... }` would be nice :)

Comment (be-ndee) [reply lua/nvim-tree/actions/file-popup.lua:33]: I also thought adding more infos into the popup, for example the file permissions. And also add the possibility to configure the displayed information.

**Pull request merged**